rskj-core/src/main/java/co/rsk/rpc/netty/http/HttpServer.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2018 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.rpc.netty.http;
+
+import co.rsk.config.InternalService;
+import co.rsk.rpc.netty.http.dto.ModuleConfigDTO;
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HttpServer implements InternalService {
+    private static final Logger logger = LoggerFactory.getLogger(HttpServer.class);
+
+    private int inetPort;
+    private ModuleConfigDTO moduleConfigDTO;
+
+    private HttpServer() { }
+
+    public HttpServer(int inetPort, ModuleConfigDTO moduleConfigDTO) {
```

Comment:
hey @nagarev, probably I’m missing something here, but QQ anyway…

why are we adding a new `HttpServer` and all related classes instead of creating a `co.rsk.rpc.modules.debug.HealthModule` or even just adding a new endpoint to `co.rsk.rpc.modules.debug.DebugModule`? what is the intention or the benefit?

Reply:
This is a functional requirement.